src/cpp/r/session/RSession.cpp

```diff
@@ -1494,14 +1494,17 @@ Error run(const ROptions& options, const RCallbacks& callbacks)
    bool loadInitFile = false;
    if (restartContext().hasSessionState())
    {
-      loadInitFile = restartContext().rProfileOnRestore();
+      loadInitFile = restartContext().rProfileOnRestore() && options.rProfileOnStart;
    }
    else
    {
-      loadInitFile = !s_suspendedSessionPath.exists()
-                     || options.rProfileOnResume
-                     || r::session::state::packratModeEnabled(
-                                                s_suspendedSessionPath);
+      // we run the .Rprofile if this is a brand new session and
+      // we are in a project and the ExecuteProfile setting is set, or we are not in a project
+      // alternatively, if we are resuming a session and the option is set to possibly run the .Rprofile
+      // we will only run it if the ExecuteProfile project setting is set (or we are not in a project)
+      loadInitFile = (!s_suspendedSessionPath.exists() && options.rProfileOnStart)
```

Comment:
Nit: could be expressed more succinctly as `options.rProfileOnStart && (!s_suspendedSessionPath.exists() || options.rProfileOnResume)`